### PR TITLE
Day5-1 to Integration

### PR DESCRIPTION
**Pull request by TeodorMatase**
I forgot to create the integration branch first but it looks like I can still merge day5-1 into integration regardless.

### CODE REVIEW

File: force-app/main/default/classes/OpportunityServiceTest.cls [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     *                             - Asserts that the accounts and opportunities exist in sf DB.
+     */
+    @TestSetup
+    private static void createTestData(){
+        List<Account> accts = TestDataFactory.createAccounts(3, true);
+        List<Opportunity> opps = TestDataFactory.createOpportunities(accts,3,true);
+        Integer numAccts = [SELECT COUNT() FROM Account WHERE Name like 'Test:%'];
+        Integer numOpps = [SELECT COUNT() FROM Opportunity WHERE Name like '%Opportunity:%'];
+        System.assertEquals(3, numAccts,'The data factory should have created 3 accounts');
+        System.assertEquals(9, numOpps, 'The data factory should hacve created 9 opportunities');
```
Comment (NatalyaMurphy) [inline force-app/main/default/classes/OpportunityServiceTest.cls:15]: Minor typo in your assert message "hacve" vs. "have"

File: force-app/main/default/classes/OpportunityServiceTest.cls [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+
+    @isTest
+    private static void getOpportunitiesById(){
+        Account acc = [SELECT Name FROM Account WHERE Name like 'Test:%' LIMIT 1];
+        List<Opportunity> opps = [SELECT Name, Amount, StageName, CloseDate, AccountId FROM Opportunity WHERE AccountId = :acc.Id ORDER BY StageName, CloseDate DESC];
```
Comment (NatalyaMurphy) [inline force-app/main/default/classes/OpportunityServiceTest.cls:22]: Use a child query here to get both the account and its opportunities at the same time.  Save your SOQL queries for when you really need them:
 SELECT Name, (SELECT Name FROM Opportunities)
 FROM Account

File: force-app/main/default/classes/OpportunityServiceTest.cls [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        System.assertEquals(true, acc.Name.contains('Test:'), 'did not pickup any account created by the testdatafactory');
+        System.assertEquals(3, opps.size(),'did not pickup any related opportunities for accountID: '+acc.Id);
```
Comment (NatalyaMurphy) [inline force-app/main/default/classes/OpportunityServiceTest.cls:24]: If using a subquery as suggested, the second parameter would change to acc.Opportunities.size()

File: force-app/main/default/classes/OpportunityServiceTest.cls [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        Test.startTest();
+        List<Opportunity> testResult = OpportunityService.getOpportunitiesByAccountId(acc.Id);
+        System.assertEquals(opps.size(), testResult.size(),'number of opportunities retrieved does not match the number of opportunities created');
```
Comment (NatalyaMurphy) [inline force-app/main/default/classes/OpportunityServiceTest.cls:27]: ditto here - change to use acc.Opportunities.size()

File: force-app/main/default/classes/TestDataFactory.cls [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+     */ 
+    public static List<Opportunity> createOpportunities(List<Account> accts, integer numOpp, boolean inserted){
+        List<Opportunity> toRet = new List<Opportunity>();
+        for(Account currentAcc:accts){
+            for(integer i = 0; i<numOpp; i++){
+                Opportunity newOpp = new Opportunity();
+                newOpp.name = currentAcc.name + ': Opportunity: '+i;
+                System.debug(newOpp.name);
+                newOpp.AccountId = currentAcc.id;
+                newOpp.StageName = 'Qualification'; //If I try prospecting I get an error because there was no Proposal_Sent__c value.
```
Comment (NatalyaMurphy) [inline force-app/main/default/classes/TestDataFactory.cls:37]: You can populate the Proposal_Sent__c in the data factory if you want

Comment (TeodorMatase) [reply force-app/main/default/classes/TestDataFactory.cls:37]: After attempting to set newOpp.Proposal_Sent__C = 'Yes' I got an "INVALID_OR_NULL_FOR_RESTRICTED_PICKLIST" error. I checked the field and it has 'Yes' as both the value and API name within the picklist so I'm not really sure what the hangup is there.

Comment (TeodorMatase) [reply force-app/main/default/classes/TestDataFactory.cls:37]: OpportunityServiceTest.getOpportunitiesById  Fail     System.DmlException: Insert failed. First exception on row 0; first error: INVALID_OR_NULL_FOR_RESTRICTED_PICKLIST, bad value for restricted picklist field: Yes: [Proposal_Sent__c]  
                                                      Class.TestDataFactory.createOpportunities: line 45, column 1                                                                                                                          
                                                      Class.OpportunityServiceTest.createTestData: line 11, column 1

**Pull request merged**